src/components/spaces/breeding/breeding-view.tsx

```diff
@@ -201,6 +243,71 @@ export class BreedingView extends BaseComponent<IProps, IState> {
     );
   }
 
+  private handleOffspringHoverEnter = (index: number) => () => {
+    this.setState({offspringHightlightIndex: index});
+  }
+  private handleOffspringHoverExit = (index: number) => () => {
+    this.setState({offspringHightlightIndex: -1});
+  }
+  private handleParentHoverEnter = (index: number) => () => {
+    this.setState({parentHightlightIndex: index});
+  }
+  private handleParentHoverExit = (index: number) => () => {
+    this.setState({parentHightlightIndex: -1});
+  }
+
+  private renderArrowPanel = (gameteLeftPositions: number[], gameteRightPositions: number[]) => {
+    const hoverArrows: ArrowInfo[] = [];
+    // calculate arrow offset due to scrolling
+    const { breeding } = this.stores;
+    const { litterSliderVal, offspringHightlightIndex } = this.state;
+    const activeBreedingPair = breeding.activeBreedingPair!;
+    const numLitters = activeBreedingPair.litters.length;
+    const sliderMax = this.getSliderMax(numLitters);
+    const sliderInterval = numLitters > 1 ? sliderMax / (numLitters - 1) : 100;
+    const sliderOffset = litterSliderVal % sliderInterval;
+    const scrollAmount = numLitters > 1
+                         ? (sliderOffset < sliderInterval / 2 ? -1 * sliderOffset : sliderInterval - sliderOffset)
+                         : 0;
+    const scrollYAdjustment = 82 * (scrollAmount / sliderInterval);
+
+    const maxGametes = 5;
+    const gameteCount = gameteLeftPositions.length;
+    const baseEndY = 46 + scrollYAdjustment;
+    const yStagger = 16;
+    const leftGameteXStart = 18;
+    const rightGameteXStart = 201;
+    const gameteXOffset = (maxGametes - gameteCount) * 10;
+    const gameteYStagger = 6;
+    const gameteDelta = 20;
+    const mouseXStart = 48;
+    const mouseXOffset = (maxGametes - gameteCount) * 24;
+    const mouseXIncrement = 48;
+    const arrowGap = 15;
+    gameteLeftPositions.forEach((position, i) => {
+      const startY = i % 2 === 1 ? gameteYStagger : 0;
+      const endY = baseEndY + (position % 2 === 1 ? yStagger : 0);
+      const startX = leftGameteXStart + gameteXOffset + i * gameteDelta;
+      const endX = mouseXStart + mouseXOffset + position * mouseXIncrement;
+      const visible = offspringHightlightIndex === position;
+      const arrow: ArrowInfo = { startX, endX, startY, endY, headRotation: -10, visible };
+      hoverArrows.push(arrow);
+    });
+    gameteRightPositions.forEach((position, i) => {
+      const startY = i % 2 === 1 ? gameteYStagger : 0;
+      const endY = baseEndY + (position % 2 === 1 ? yStagger : 0);
+      const startX = rightGameteXStart + gameteXOffset + i * gameteDelta;
+      const endX = mouseXStart + mouseXOffset + position * mouseXIncrement + arrowGap;
+      const visible = offspringHightlightIndex === position;
+      const arrow: ArrowInfo = { startX, endX, startY, endY, headRotation: 10, visible };
+      hoverArrows.push(arrow);
+    });
```

Comment:
This all currently works very well. I worry a little bit about how brittle it is. Did you consider getting the element position directly from the DOM?

It might have been possible to simply attach a `ref` to the elements in the gametes and the current litter, and then use `.getBoundingClientRect()` to get the exact position of the arrow start and ends relative to the viewport. Then this ought to just work even if we move the offspring images around.

But again, this does work, so maybe there's no need to change it, unless perhaps we find ourselves going back to make tweaks in the future.

Reply:
Agreed.  It is brittle.  I originally planned to try to access the positions by grabbing elements out of the DOM, but it felt like more work than it was worth so I went this route of baking in a handful of constants and doing a little math.  Maybe I'm wrong, however.  It might not be that bad to use a React ref to then pull the exact of the components to get the start/end locations needed to build the arrows.  

At this point in the sprint, I would vote that we merge this in for now, and I'll try to circle back to it in the next 48 hours if time permits (or add a refactor story to PT).  What do you think of a plan like that?

Reply:
:+1: